packages/react/src/components/SideMenu/SideMenuController.tsx

```diff
@@ -47,56 +55,97 @@ export const SideMenuController = (props: {
                 return;
               }
 
-              if (block.type === "heading") {
-                if (!block.props.level || block.props.level === 1) {
-                  elements.floating.style.setProperty("height", "78px");
-                  return;
-                }
+              const blockElement =
+                elements.reference instanceof Element
+                  ? elements.reference
+                  : elements.reference.contextElement;
+              if (blockElement === undefined) {
+                return;
+              }
 
-                if (block.props.level === 2) {
-                  elements.floating.style.setProperty("height", "54px");
-                  return;
-                }
+              const blockContentElement =
+                blockElement.querySelector(".bn-block-content");
+              if (blockContentElement === null) {
+                return;
+              }
+
+              const blockContentBoundingClientRect =
+                blockContentElement.getBoundingClientRect();
+
+              // Special case for file blocks with captions. In this case, we
+              // align the side menu with the first sibling of the file caption
+              // element.
+              const fileCaptionParentElement =
```

Comment:
imo we can't have these special cases hardcoded here. The sidemenu should not be aware of our block types or their HTML structure. Can we come up with a more future-proof API?

Also, is it easy to add e2e tests for this to prevent regressions?

Reply:
what about a classname or other child element selector that we can query for? (unless @nperez0111 can come up with something cleaner)?

Imo most important is that this component should imo have no knowledge of different block types and their implementations

Reply:
I thought the logic would look like:

```ts
const exceptions = [ ".file-caption", ".xyz" ]

let subtractFromHeight = 0;

exceptions.forEach(e=>{
subtractFromHeight += document.querySelector(e)?.getBoundingBox().height ?? 0
})
```

But, it's unclear to me whether @matthewlipski is doing any sort of alignment of buttons or not.

If it were just to subtract things from being counted as the height then either this, or just introducing a classname to put on any element we want to be discounted from the height would work like @YousefED mentioned.